--- twinkle-1.11.0.orig/cmake/FindIlbc.cmake	2026-09-14 21:32:43.000000000 +0200
+++ twinkle-1.11.0/cmake/FindIlbc.cmake	2026-09-15 15:31:51.895844350 +0200
@@ -1,7 +1,7 @@
 include (CMakePushCheckState)
 include (CheckCXXSourceCompiles)
 
-FIND_PATH(ILBC_INCLUDE_DIR ilbc/iLBC_decode.h)
+FIND_PATH(ILBC_INCLUDE_DIR ilbc.h)
 FIND_LIBRARY(ILBC_LIBRARIES NAMES ilbc)
 
 IF(ILBC_INCLUDE_DIR AND ILBC_LIBRARIES)
@@ -13,7 +13,7 @@
 	LIST(APPEND CMAKE_REQUIRED_LIBRARIES "-lilbc")
 	SET(CMAKE_REQUIRED_QUIET TRUE)
 	CHECK_CXX_SOURCE_COMPILES("
-		#include <ilbc/iLBC_decode.h>
+		#include <ilbc.h>
 
 		int main() {
 			iLBC_Dec_Inst_t *iLBCdec_inst;
@@ -26,7 +26,7 @@
 
 IF(ILBC_FOUND)
 	IF (NOT Ilbc_FIND_QUIETLY)
-		MESSAGE(STATUS "Found ilbc includes:	${ILBC_INCLUDE_DIR}/ilbc/iLBC_decode.h")
+		MESSAGE(STATUS "Found ilbc includes:	${ILBC_INCLUDE_DIR}/ilbc.h")
 		MESSAGE(STATUS "Found ilbc library: ${ILBC_LIBRARIES}")
 	ENDIF (NOT Ilbc_FIND_QUIETLY)
 ELSE(ILBC_FOUND)
--- twinkle-1.11.0.orig/src/audio/audio_decoder.cpp	2026-09-14 21:32:43.000000000 +0200
+++ twinkle-1.11.0/src/audio/audio_decoder.cpp	2026-09-15 15:32:24.347167605 +0200
@@ -24,7 +24,13 @@
 #ifndef HAVE_ILBC_CPP
 extern "C" {
 #endif
-#include <ilbc/iLBC_decode.h>
+#include <ilbc.h>
+#ifndef NO_OF_BYTES_20MS
+#define NO_OF_BYTES_20MS 38
+#define NO_OF_BYTES_30MS 50
+#define MIN_SAMPLE -32768
+#define MAX_SAMPLE 32767
+#endif
 #ifndef HAVE_ILBC_CPP
 }
 #endif
@@ -278,18 +284,18 @@
 		int16 *pcm_buf, uint16 pcm_buf_size)
 {
 	float sample;
-	float block[BLOCKL_MAX];
+	int16_t block[BLOCKL_MAX];
 	int block_len;
 	
 	if (get_ptime(payload_size) == 20) {
 		block_len = BLOCKL_20MS;
 		assert(pcm_buf_size >= block_len);
-		iLBC_decode(block, (unsigned char*)payload, &_ilbc_decoder_20, 1);
+		iLBC_decode(block, (uint16*)payload, &_ilbc_decoder_20, 1);
 		_last_received_ptime = 20;
 	} else {
 		block_len = BLOCKL_30MS;
 		assert(pcm_buf_size >= block_len);
-		iLBC_decode(block, (unsigned char*)payload, &_ilbc_decoder_30, 1);
+		iLBC_decode(block, (uint16*)payload, &_ilbc_decoder_30, 1);
 		_last_received_ptime = 30;
 	}
 	
@@ -306,8 +312,8 @@
 }
 
 uint16 t_ilbc_audio_decoder::conceal(int16 *pcm_buf, uint16 pcm_buf_size) {
-	float sample;
-	float block[BLOCKL_MAX];
+	int16 sample;
+	int16 block[BLOCKL_MAX];
 	int block_len;
 	
 	if (_last_received_ptime == 0) return 0;
--- twinkle-1.11.0.orig/src/audio/audio_decoder.h	2026-09-14 21:32:43.000000000 +0200
+++ twinkle-1.11.0/src/audio/audio_decoder.h	2026-09-15 15:31:51.885631651 +0200
@@ -52,7 +52,7 @@
 #ifndef HAVE_ILBC_CPP
 extern "C" {
 #endif
-#include <ilbc/iLBC_define.h>
+#include <ilbc.h>
 #ifndef HAVE_ILBC_CPP
 }
 #endif
--- twinkle-1.11.0.orig/src/audio/audio_encoder.cpp	2026-09-14 21:32:43.000000000 +0200
+++ twinkle-1.11.0/src/audio/audio_encoder.cpp	2026-09-15 15:32:24.340059253 +0200
@@ -23,7 +23,11 @@
 #ifndef HAVE_ILBC_CPP
 extern "C" {
 #endif
-#include <ilbc/iLBC_encode.h>
+#include <ilbc.h>
+#ifndef NO_OF_BYTES_20MS
+#define NO_OF_BYTES_20MS 38
+#define NO_OF_BYTES_30MS 50
+#endif
 #ifndef HAVE_ILBC_CPP
 }
 #endif
@@ -271,13 +275,8 @@
 	assert(nsamples == _ilbc_encoder.blockl);
 	
 	silence = false;
-	float block[nsamples];
-	
-	for (int i = 0; i < nsamples; i++) {
-		block[i] = static_cast<float>(sample_buf[i]);
-	}
 	
-	iLBC_encode((unsigned char*)payload, block, &_ilbc_encoder);
+	iLBC_encode((uint16*)payload, sample_buf, &_ilbc_encoder);
 	
 	return _ilbc_encoder.no_of_bytes;
 }
--- twinkle-1.11.0.orig/src/audio/audio_encoder.h	2026-09-14 21:32:43.000000000 +0200
+++ twinkle-1.11.0/src/audio/audio_encoder.h	2026-09-15 15:31:51.885631651 +0200
@@ -51,7 +51,7 @@
 #ifndef HAVE_ILBC_CPP
 extern "C" {
 #endif
-#include <ilbc/iLBC_define.h>
+#include <ilbc.h>
 #ifndef HAVE_ILBC_CPP
 }
 #endif
